bootstrap/v2/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -205,28 +206,46 @@ func (kustomize *kustomize) Apply(resources kftypes.ResourceEnum) error {
 
 	kustomizeDir := path.Join(kustomize.Spec.AppDir, outputDir)
 	for _, compName := range kustomize.Spec.Components {
-		kustomizeFile := filepath.Join(kustomizeDir, compName+".yaml")
-		if _, err := os.Stat(kustomizeFile); err != nil {
+		resMap, err := GenerateKustomizationFile(&kustomize.KfDef, kustomizeDir, compName,
```

Comment:
Is GenerateKustomizationFile creating the kustomizaiton.YAML file? Or is it creating the kustomization.yaml file?

It looks like its creating the kustomizationFile because its taking ComponentParams.

Should that be done in generate?

The idea would be that if a user runs
kfctl generate
edits the kustomize package
kfctl apply

Then any user supplied edits should be picked up.

But right now it looks like those changes would be overwritten if during apply we call GenerateKustomizationFile.

Not sure that's actually possible right now with the way GenerateKustomizationFile works; it looks like its coupling the writing of the kustomization file with evaluating overlays and params.

What if we just ran GenerateKustomizationFile during generate like we do now but the Copying of the
kustomize package so that kustomize/  would contain the list of kustomize packages.

Reply:
makes sense, thanks.

I separated the writing kustomize.yaml and evaluation part in GenerateKustomizationFile